tests/integration/tests/remote_istiod_test.go

```diff
@@ -70,6 +70,8 @@ func TestRemoteIstiod(t *testing.T) {
 		}
 		_, err = kubeClient.AppsV1().Deployments(originalConf.IstioNamespace).Update(ctx, istiod, metav1.UpdateOptions{})
 		require.NoError(err)
+
+		require.NoError(kube.WaitForDeploymentReady(ctx, kubeClient, originalConf.IstioNamespace, istioDeploymentName))
```

Comment:
This is the whole change. The rest is moving the `WaitForDeploymentReady` so it can be exported and reused.